torchtitan/experiments/llama4/model/moe.py

```diff
@@ -209,6 +213,35 @@ def __init__(self, model_args: TransformerModelArgs):
             else None
         )
 
+        # load balancing
+        self.load_balance_coeff = model_args.load_balance_coeff
+        # the fields below are defined even when load_balance_coeff is None
+        # to make initialization and checkpointing code simpler
+        self.register_buffer(
+            "expert_bias",
+            torch.zeros(num_experts, dtype=torch.float32),
+            persistent=True,
+        )
+        self.register_buffer(
+            "tokens_per_expert",
+            torch.zeros(num_experts, dtype=torch.float32),
+            persistent=True,
+        )
+
+        # NOTE: forward hook, forward pre hook, or backward pre hook
+        #       would conflict with activation checkpointing
```

Comment:
uh, this make sense that `backward pre hook` would conflict. Didn't think about it through.